compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -144,9 +144,15 @@ object Inliner {
           case _: Apply | _: TypeApply | _: RefTree if needsAccessor(tree.symbol) =>
             if (tree.isTerm) {
               val (methPart, targs, argss) = decomposeCall(tree)
-              addAccessor(tree, methPart, targs, argss,
-                  accessedType = methPart.tpe.widen,
-                  rhs = (qual, tps, argss) => qual.appliedToTypes(tps).appliedToArgss(argss))
+              val mSym = methPart.symbol
+              if (mSym.isConstructor && (mSym.is(Private) || mSym.privateWithin.exists)) {
```

Comment:
I thought so too but when trying it out I found that it was not the case. Should it be? Did I stumble on another bug?